scripts/test-w-coverage.sh

```diff
@@ -2,15 +2,25 @@
 
 set -aueo pipefail
 
-readarray -t modules < <(go list ./... | grep -v tests/framework | grep -v tests/e2e | grep -v tests/scenarios | grep -v tests/scale | \
-   grep -v ci/ | grep -v demo/ | grep -v experimental/ | grep -v scripts/)
+readarray -t modules < <(go list ./... | \
+                             grep -v tests/framework | \
+                             grep -v tests/e2e | \
+                             grep -v tests/scenarios | \
+                             grep -v tests/scale | \
+                             grep -v ci/ | \
+                             grep -v demo/ | \
+                             grep -v experimental/ | \
+                             grep -v scripts/)
 
 go test -timeout 120s \
    -failfast \
    -v \
-   -coverprofile=coverage.txt \
+   -coverprofile=coverage.txt.with_generated_code \
    -covermode count "${modules[@]}" | tee testoutput.txt || { echo "go test returned non-zero"; exit 1; }
 
+# shellcheck disable=SC2002
+cat coverage.txt.with_generated_code | grep -v "_generated.go" > coverage.txt
+
 # shellcheck disable=SC2002
 cat testoutput.txt | go run github.com/jstemmer/go-junit-report > report.xml
```

Comment:
Do we need to regenerate `testoutput.txt` or is this line not required at all?
Also doesn't `go tool cover` need to be run post stripping the autogen files?

Reply:
We had another tool use report.xml at some point. Not sure it is in use - I'll be inclined to leave it.

The `go test -cover` tool will create this (in `coverage.txt.with_generated_code`):
```
github.com/openservicemesh/osm/pkg/smi/fake.go:74.83,76.2 1 0
github.com/openservicemesh/osm/pkg/smi/mock_meshspec_generated.go:29.61,33.2 3 0
github.com/openservicemesh/osm/pkg/smi/mock_meshspec_generated.go:36.59,38.2 1 0
```

Then we remove `_generated.go` lines:
```
github.com/openservicemesh/osm/pkg/smi/fake.go:74.83,76.2 1 0
```
and that goes into `coverage.txt`
 and then we run
`go run github.com/axw/gocov/gocov convert coverage.txt > coverage.json`
and finally `go run github.com/matm/gocov-html < coverage.json > coverage/index.html`, which is what I use (the HTML file)

Reply:
As long as you see coverage improve, whatever works!